llvm/lib/Transforms/Scalar/JumpThreading.cpp

```diff
@@ -1362,11 +1362,14 @@ bool JumpThreadingPass::simplifyPartiallyRedundantLoad(LoadInst *LoadI) {
   // farther than to a predecessor, we need to reuse the code from GVN's PRE.
   // It requires domination tree analysis, so for this simple case it is an
   // overkill.
+  bool TransfersExecution = false;
   if (PredsScanned.size() != AvailablePreds.size() &&
-      !isSafeToSpeculativelyExecute(LoadI))
+      !isSafeToSpeculativelyExecute(LoadI)) {
     for (auto I = LoadBB->begin(); &*I != LoadI; ++I)
       if (!isGuaranteedToTransferExecutionToSuccessor(&*I))
         return false;
+    TransfersExecution = true;
```

Comment:
I think we need to always compute TransfersExecution here (even if not safe to speculate), otherwise we'll get the weird situation where marking the load dereferenceable will result in less metadata transfer.

Reply:
Fixed.